ENIGMAsystem/SHELL/Universal_System/Resources/fontstruct.cpp

```diff
@@ -224,6 +225,12 @@ string font_get_fontname(int fnt)
 
 void font_delete(int fnt)
 {
+    if (enigma::currentfont == fnt) {
+       enigma::currentfont = -1;
+       #ifdef DEBUG_MODE
+       DEBUG_MESSAGE("Deleted currently in use font " + std::to_string(fnt) , MESSAGE_TYPE::M_USER_ERROR);
+       #endif
+    }
```

Comment:
Here is an alternative that works if something else happens besides a delete.
```cpp
struct CurrentFont {
  CurrentFont(int id): _id(id) {}
  operator=(int other) { _id = other; }
  operator int() { 
    if (!sprite_fonts.exists(_id)) return -1;
    return _id;
  }
} currentfont;
```

Reply:
Ok, so in every other respect this pull request is just like GMSv1.4 except for font deletion and possibly the current font. I don't think it's a good idea that we allow deletion of the default font... but I suppose we can allow it. However, GMSv1.4 does not allow it, just like all of the other functions which accept a font id.

![GMSv1.4 Delete Default Font Error](https://user-images.githubusercontent.com/3212801/89819431-73b30980-db19-11ea-983f-368a53cdaefb.png)

Reply:
I made a little test for this specifically. Just creating a new font and setting its size to 56 before running. It does not produce a debug message in GMSv1.4 like it does this pull request. However, it does show that GM properly recognizes the current font being deleted and updates accordingly.
```cpp
draw_set_font(font0);
show_message(string(string_width("hellooooooo"))); // 412
font_delete(font0);
show_message(string(string_width("hellooooooo"))); // 102
```